// loading-animations-react/dist/main.js
require("./main.css");
var $cvI92$reactjsxruntime = require("react/jsx-runtime");
require("react");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "Dots", function () { return $8b0d76cc85c8a2f5$export$8da50d3722682be0; });
$parcel$export(module.exports, "MinimalSpinner", function () { return $11ca252ee044ba7d$export$7248a4fc0dcfadde; });
$parcel$export(module.exports, "ProgressBar", function () { return $9238cdd8f388646b$export$c17561cb55d4db30; });
$parcel$export(module.exports, "Spinner", function () { return $ff2eab4d786ad98f$export$7f7cbe89f1eacd2; });
$parcel$export(module.exports, "TrinitySpinner", function () { return $5c7c07324c275910$export$7bee1434560ae9ef; });
$parcel$export(module.exports, "Waves", function () { return $ed86c5eef54fce5b$export$3f008f9f514cfb66; });


var $bad8531e74e9c8fc$export$31a76fba2315ab58;
var $bad8531e74e9c8fc$export$e5383273bc7bedd;
var $bad8531e74e9c8fc$export$94132a0e348806d4;
var $bad8531e74e9c8fc$export$f4c5de44377d2946;
var $bad8531e74e9c8fc$export$e189116d13f2853d;
$bad8531e74e9c8fc$export$31a76fba2315ab58 = "_index__dotsWrapper";
$bad8531e74e9c8fc$export$e5383273bc7bedd = "_index__dots";
$bad8531e74e9c8fc$export$94132a0e348806d4 = "_index__dot";
$bad8531e74e9c8fc$export$f4c5de44377d2946 = "_index__loader";
$bad8531e74e9c8fc$export$e189116d13f2853d = "_index__dotsText";


function $8b0d76cc85c8a2f5$export$8da50d3722682be0(param) {
    var _dotColors = param.dotColors, dotColors = _dotColors === void 0 ? [
        'blue',
        'violet',
        'red',
        'orange',
        'yellow',
        'green'
    ] : _dotColors, _text = param.text, text = _text === void 0 ? 'Loading...' : _text, _id = param.id, id = _id === void 0 ? '' : _id, _className = param.className, className = _className === void 0 ? '' : _className;
    // Setup newColors array
    var newColors = [];
    var dotColorsIndex = 0;
    var newColorsIndex = 0;
    while(newColorsIndex <= 5)if (dotColors[dotColorsIndex]) {
        newColors.push(dotColors[dotColorsIndex]);
        newColorsIndex += 1;
        dotColorsIndex += 1;
    } else dotColorsIndex = 0;
     // Setup array of dot elements
    var dots = new Array(6).fill('').map(function(e, index) {
        return(/*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
            className: $bad8531e74e9c8fc$export$94132a0e348806d4,
            style: {
                backgroundColor: newColors[index],
                animationDelay: "0.".concat(index, "s")
            }
        }, index));
    });
    return(/*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("div", {
        id: id,
        className: "".concat($bad8531e74e9c8fc$export$31a76fba2315ab58, " ").concat(className),
        children: [
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                className: $bad8531e74e9c8fc$export$e5383273bc7bedd,
                children: dots
            }),
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                className: $bad8531e74e9c8fc$export$e189116d13f2853d,
                children: text
            })
        ]
    }));
}



var $a1035865e3d2aa55$export$9d2315782d9ad1b7;
var $a1035865e3d2aa55$export$8b169d9a7a6c1f1e;
var $a1035865e3d2aa55$export$e5c9b4239c4e8822;
var $a1035865e3d2aa55$export$f8f398eee63e254;
var $a1035865e3d2aa55$export$3698fbbd6e8d1037;
$a1035865e3d2aa55$export$9d2315782d9ad1b7 = "_index__minimalSpinnerWrapper";
$a1035865e3d2aa55$export$8b169d9a7a6c1f1e = "_index__minimalSpinner";
$a1035865e3d2aa55$export$e5c9b4239c4e8822 = "_index__rotate-spinner";
$a1035865e3d2aa55$export$f8f398eee63e254 = "_index__minimalSpinnerText";
$a1035865e3d2aa55$export$3698fbbd6e8d1037 = "_index__rotate-text";


function $11ca252ee044ba7d$export$7248a4fc0dcfadde(param) {
    var _text = param.text, text = _text === void 0 ? 'Loading...' : _text, _color = param.color, color = _color === void 0 ? 'blue' : _color, _id = param.id, id = _id === void 0 ? '' : _id, _className = param.className, className = _className === void 0 ? '' : _className;
    var style = "\n    .".concat($a1035865e3d2aa55$export$8b169d9a7a6c1f1e, " {\n      border-color: transparent ").concat(color, " transparent ").concat(color, ";\n    }\n    .").concat($a1035865e3d2aa55$export$f8f398eee63e254, " {\n      color: ").concat(color, ";\n    }");
    return(/*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("div", {
        id: id,
        className: "".concat($a1035865e3d2aa55$export$9d2315782d9ad1b7, " ").concat(className),
        children: [
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                className: $a1035865e3d2aa55$export$8b169d9a7a6c1f1e,
                children: /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                    className: $a1035865e3d2aa55$export$f8f398eee63e254,
                    children: text
                })
            }),
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("style", {
                children: style
            })
        ]
    }));
}



var $fa30838016cc88ae$export$43167351f6db920a;
var $fa30838016cc88ae$export$9d1298853f0e77f7;
var $fa30838016cc88ae$export$efb91eddb63d470a;
var $fa30838016cc88ae$export$b38bf4f33e9e4c25;
var $fa30838016cc88ae$export$238004eea37cd0e4;
var $fa30838016cc88ae$export$1f9a16440fb0b423;
$fa30838016cc88ae$export$43167351f6db920a = "_index__progressBarWrapper";
$fa30838016cc88ae$export$9d1298853f0e77f7 = "_index__progressBar";
$fa30838016cc88ae$export$efb91eddb63d470a = "_index__sliderBackground";
$fa30838016cc88ae$export$b38bf4f33e9e4c25 = "_index__slider";
$fa30838016cc88ae$export$238004eea37cd0e4 = "_index__progres";
$fa30838016cc88ae$export$1f9a16440fb0b423 = "_index__progressBarText";


function $9238cdd8f388646b$export$c17561cb55d4db30(param) {
    var _text = param.text, text = _text === void 0 ? 'Loading...' : _text, _borderColor = param.borderColor, borderColor = _borderColor === void 0 ? 'navy' : _borderColor, _sliderColor = param.sliderColor, sliderColor = _sliderColor === void 0 ? 'white' : _sliderColor, _sliderBackground = param.sliderBackground, sliderBackground = _sliderBackground === void 0 ? 'red' : _sliderBackground, _id = param.id, id = _id === void 0 ? '' : _id, _className = param.className, className = _className === void 0 ? '' : _className;
    var style = "\n    .".concat($fa30838016cc88ae$export$9d1298853f0e77f7, " {\n      background-color: ").concat(borderColor, ";\n    }\n    .").concat($fa30838016cc88ae$export$efb91eddb63d470a, " {\n      background-color: ").concat(sliderBackground, ";\n    }\n    .").concat($fa30838016cc88ae$export$b38bf4f33e9e4c25, " {\n      background-color: ").concat(sliderColor, ";\n    }");
    return(/*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("div", {
        id: id,
        className: "".concat($fa30838016cc88ae$export$43167351f6db920a, " ").concat(className),
        children: [
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                className: $fa30838016cc88ae$export$9d1298853f0e77f7,
                children: /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                    className: $fa30838016cc88ae$export$efb91eddb63d470a,
                    children: /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                        className: $fa30838016cc88ae$export$b38bf4f33e9e4c25
                    })
                })
            }),
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                className: $fa30838016cc88ae$export$1f9a16440fb0b423,
                children: text
            }),
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("style", {
                children: style
            })
        ]
    }));
}



var $2b688e894453d86e$export$d043ff22bb17d686;
var $2b688e894453d86e$export$641374ffb95bc399;
var $2b688e894453d86e$export$a179c88182cb9d56;
var $2b688e894453d86e$export$bb628a54ab399bc9;
var $2b688e894453d86e$export$e1d786d2f707b414;
var $2b688e894453d86e$export$eb03490a10c3abc9;
var $2b688e894453d86e$export$6f5fc7f68f30286;
$2b688e894453d86e$export$d043ff22bb17d686 = "_index__spinnerWrapper";
$2b688e894453d86e$export$641374ffb95bc399 = "_index__spinner";
$2b688e894453d86e$export$a179c88182cb9d56 = "_index__viewBox";
$2b688e894453d86e$export$bb628a54ab399bc9 = "_index__rotate";
$2b688e894453d86e$export$e1d786d2f707b414 = "_index__circle";
$2b688e894453d86e$export$eb03490a10c3abc9 = "_index__dash";
$2b688e894453d86e$export$6f5fc7f68f30286 = "_index__spinnerText";


function $ff2eab4d786ad98f$export$7f7cbe89f1eacd2(param) {
    var _id = param.id, id = _id === void 0 ? '' : _id, _className = param.className, className = _className === void 0 ? '' : _className, _color1 = param.color1, color1 = _color1 === void 0 ? 'green' : _color1, _color2 = param.color2, color2 = _color2 === void 0 ? 'lightGreen' : _color2, _text = param.text, text = _text === void 0 ? 'Loading...' : _text, _textColor = param.textColor, textColor = _textColor === void 0 ? '' : _textColor;
    var style = ".".concat($2b688e894453d86e$export$6f5fc7f68f30286, " {\n    background: linear-gradient(90deg, ").concat(color1, ", ").concat(color2, ");\n    background-clip: text;\n    -webkit-background-clip: text;\n    -webkit-text-fill-color: transparent;\n    font-weight: bold;\n  }");
    if (textColor !== '') style = "\n      .".concat($2b688e894453d86e$export$6f5fc7f68f30286, " {\n      font-weight: bold;\n      color: ").concat(textColor, "\n      }");
    return(/*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("div", {
        id: id,
        className: "".concat($2b688e894453d86e$export$d043ff22bb17d686, " ").concat(className),
        children: [
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("div", {
                className: $2b688e894453d86e$export$641374ffb95bc399,
                children: [
                    /*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("svg", {
                        className: $2b688e894453d86e$export$a179c88182cb9d56,
                        viewBox: "0 0 50 50",
                        children: [
                            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("circle", {
                                className: $2b688e894453d86e$export$e1d786d2f707b414,
                                cx: "25",
                                cy: "25",
                                r: "22.5",
                                fill: "none",
                                strokeWidth: "5",
                                stroke: "url(#myGradient)"
                            }),
                            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("defs", {
                                children: /*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("linearGradient", {
                                    id: "myGradient",
                                    children: [
                                        /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("stop", {
                                            offset: "0%",
                                            stopColor: color1
                                        }),
                                        /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("stop", {
                                            offset: "100%",
                                            stopColor: color2
                                        })
                                    ]
                                })
                            })
                        ]
                    }),
                    /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                        className: "gradientText ".concat($2b688e894453d86e$export$6f5fc7f68f30286),
                        children: text
                    })
                ]
            }),
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("style", {
                children: style
            })
        ]
    }));
}



var $30b4eadc4220d4c7$export$77ff515d9de063ae;
var $30b4eadc4220d4c7$export$ba2737e33880d681;
var $30b4eadc4220d4c7$export$641374ffb95bc399;
var $30b4eadc4220d4c7$export$e1d786d2f707b414;
var $30b4eadc4220d4c7$export$ecad260a8a5fef4f;
var $30b4eadc4220d4c7$export$71d7c553ee594ba2;
$30b4eadc4220d4c7$export$77ff515d9de063ae = "_index__trinitySpinnerWrapper";
$30b4eadc4220d4c7$export$ba2737e33880d681 = "_index__trinitySpinner";
$30b4eadc4220d4c7$export$641374ffb95bc399 = "_index__spinner";
$30b4eadc4220d4c7$export$e1d786d2f707b414 = "_index__circle";
$30b4eadc4220d4c7$export$ecad260a8a5fef4f = "_index__animation";
$30b4eadc4220d4c7$export$71d7c553ee594ba2 = "_index__trinitySpinnerText";


function $5c7c07324c275910$export$7bee1434560ae9ef(param) {
    var _color = param.color, color = _color === void 0 ? 'red' : _color, _text = param.text, text = _text === void 0 ? 'Loading...' : _text, _id = param.id, id = _id === void 0 ? '' : _id, _className = param.className, className = _className === void 0 ? '' : _className;
    var style = ".".concat($30b4eadc4220d4c7$export$71d7c553ee594ba2, " { color: ").concat(color, " }");
    return(/*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("div", {
        id: id,
        className: "".concat($30b4eadc4220d4c7$export$77ff515d9de063ae, " ").concat(className),
        children: [
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("div", {
                className: $30b4eadc4220d4c7$export$ba2737e33880d681,
                children: [
                    /*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("svg", {
                        className: $30b4eadc4220d4c7$export$641374ffb95bc399,
                        viewBox: "0 0 100 100",
                        children: [
                            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("defs", {
                                children: /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("filter", {
                                    id: "shadow",
                                    children: /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("feDropShadow", {
                                        dx: "0",
                                        dy: "0",
                                        stdDeviation: "2",
                                        floodColor: "black"
                                    })
                                })
                            }),
                            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("circle", {
                                className: $30b4eadc4220d4c7$export$e1d786d2f707b414,
                                style: {
                                    fill: 'transparent',
                                    stroke: color,
                                    strokeWidth: '7px',
                                    strokeLinecap: 'round',
                                    filter: 'url(#shadow)'
                                },
                                cx: "50",
                                cy: "50",
                                r: "45"
                            })
                        ]
                    }),
                    /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                        className: $30b4eadc4220d4c7$export$71d7c553ee594ba2,
                        children: text
                    })
                ]
            }),
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("style", {
                children: style
            })
        ]
    }));
}



var $8a0e20071187f529$export$296d15118585a768;
var $8a0e20071187f529$export$13c70a3bcea7ce2;
var $8a0e20071187f529$export$3af1cb7633944de2;
var $8a0e20071187f529$export$6c4586e4e8bd4b48;
$8a0e20071187f529$export$296d15118585a768 = "_index__wavesWrapper";
$8a0e20071187f529$export$13c70a3bcea7ce2 = "_index__waves";
$8a0e20071187f529$export$3af1cb7633944de2 = "_index__wave";
$8a0e20071187f529$export$6c4586e4e8bd4b48 = "_index__wavesText";


function $ed86c5eef54fce5b$export$3f008f9f514cfb66(param) {
    var _backgroundColor = param.backgroundColor, backgroundColor = _backgroundColor === void 0 ? '#000' : _backgroundColor, _waveColor = param.waveColor, waveColor = _waveColor === void 0 ? 'cyan' : _waveColor, _text = param.text, text = _text === void 0 ? 'Loading...' : _text, _id = param.id, id = _id === void 0 ? '' : _id, _className = param.className, className = _className === void 0 ? '' : _className;
    var style = "\n    .".concat($8a0e20071187f529$export$13c70a3bcea7ce2, " {\n      background-color: ").concat(backgroundColor, "\n    }\n    .").concat($8a0e20071187f529$export$3af1cb7633944de2, " {\n      background: linear-gradient(45deg, ").concat(waveColor, ", #fff)\n    }");
    // Setup array of wave elements
    var waves = new Array(10).fill('').map(function(e, index) {
        return(/*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
            className: $8a0e20071187f529$export$3af1cb7633944de2,
            style: {
                animationDelay: "0.".concat(index, "s")
            }
        }, index));
    });
    return(/*#__PURE__*/ $cvI92$reactjsxruntime.jsxs("div", {
        id: id,
        className: "".concat($8a0e20071187f529$export$296d15118585a768, " ").concat(className),
        children: [
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                className: $8a0e20071187f529$export$13c70a3bcea7ce2,
                children: waves
            }),
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("div", {
                className: $8a0e20071187f529$export$6c4586e4e8bd4b48,
                children: text
            }),
            /*#__PURE__*/ $cvI92$reactjsxruntime.jsx("style", {
                children: style
            })
        ]
    }));
}
