// loading-animations-react/dist/module.js
import "./main.css";
import {jsx as $jKrDH$jsx, jsxs as $jKrDH$jsxs} from "react/jsx-runtime";
import "react";



var $140a52bee44a6d3e$export$31a76fba2315ab58;
var $140a52bee44a6d3e$export$e5383273bc7bedd;
var $140a52bee44a6d3e$export$94132a0e348806d4;
var $140a52bee44a6d3e$export$f4c5de44377d2946;
var $140a52bee44a6d3e$export$e189116d13f2853d;
$140a52bee44a6d3e$export$31a76fba2315ab58 = "_index__dotsWrapper";
$140a52bee44a6d3e$export$e5383273bc7bedd = "_index__dots";
$140a52bee44a6d3e$export$94132a0e348806d4 = "_index__dot";
$140a52bee44a6d3e$export$f4c5de44377d2946 = "_index__loader";
$140a52bee44a6d3e$export$e189116d13f2853d = "_index__dotsText";


function $f2353fd21d89f897$export$8da50d3722682be0({ dotColors: dotColors = [
    'blue',
    'violet',
    'red',
    'orange',
    'yellow',
    'green'
] , text: text = 'Loading...' , id: id = '' , className: className = '' ,  }) {
    // Setup newColors array
    const newColors = [];
    let dotColorsIndex = 0;
    let newColorsIndex = 0;
    while(newColorsIndex <= 5)if (dotColors[dotColorsIndex]) {
        newColors.push(dotColors[dotColorsIndex]);
        newColorsIndex += 1;
        dotColorsIndex += 1;
    } else dotColorsIndex = 0;
     // Setup array of dot elements
    let dots = new Array(6).fill('').map((e, index)=>{
        return(/*#__PURE__*/ $jKrDH$jsx("div", {
            className: $140a52bee44a6d3e$export$94132a0e348806d4,
            style: {
                backgroundColor: newColors[index],
                animationDelay: `0.${index}s`
            }
        }, index));
    });
    return(/*#__PURE__*/ $jKrDH$jsxs("div", {
        id: id,
        className: `${$140a52bee44a6d3e$export$31a76fba2315ab58} ${className}`,
        children: [
            /*#__PURE__*/ $jKrDH$jsx("div", {
                className: $140a52bee44a6d3e$export$e5383273bc7bedd,
                children: dots
            }),
            /*#__PURE__*/ $jKrDH$jsx("div", {
                className: $140a52bee44a6d3e$export$e189116d13f2853d,
                children: text
            })
        ]
    }));
}



var $cfe122cfe31841c3$export$9d2315782d9ad1b7;
var $cfe122cfe31841c3$export$8b169d9a7a6c1f1e;
var $cfe122cfe31841c3$export$e5c9b4239c4e8822;
var $cfe122cfe31841c3$export$f8f398eee63e254;
var $cfe122cfe31841c3$export$3698fbbd6e8d1037;
$cfe122cfe31841c3$export$9d2315782d9ad1b7 = "_index__minimalSpinnerWrapper";
$cfe122cfe31841c3$export$8b169d9a7a6c1f1e = "_index__minimalSpinner";
$cfe122cfe31841c3$export$e5c9b4239c4e8822 = "_index__rotate-spinner";
$cfe122cfe31841c3$export$f8f398eee63e254 = "_index__minimalSpinnerText";
$cfe122cfe31841c3$export$3698fbbd6e8d1037 = "_index__rotate-text";


function $763dec067a0c5467$export$7248a4fc0dcfadde({ text: text = 'Loading...' , color: color = 'blue' , id: id = '' , className: className = '' ,  }) {
    const style = `
    .${$cfe122cfe31841c3$export$8b169d9a7a6c1f1e} {
      border-color: transparent ${color} transparent ${color};
    }
    .${$cfe122cfe31841c3$export$f8f398eee63e254} {
      color: ${color};
    }`;
    return(/*#__PURE__*/ $jKrDH$jsxs("div", {
        id: id,
        className: `${$cfe122cfe31841c3$export$9d2315782d9ad1b7} ${className}`,
        children: [
            /*#__PURE__*/ $jKrDH$jsx("div", {
                className: $cfe122cfe31841c3$export$8b169d9a7a6c1f1e,
                children: /*#__PURE__*/ $jKrDH$jsx("div", {
                    className: $cfe122cfe31841c3$export$f8f398eee63e254,
                    children: text
                })
            }),
            /*#__PURE__*/ $jKrDH$jsx("style", {
                children: style
            })
        ]
    }));
}



var $d1ef67c10ada5672$export$43167351f6db920a;
var $d1ef67c10ada5672$export$9d1298853f0e77f7;
var $d1ef67c10ada5672$export$efb91eddb63d470a;
var $d1ef67c10ada5672$export$b38bf4f33e9e4c25;
var $d1ef67c10ada5672$export$238004eea37cd0e4;
var $d1ef67c10ada5672$export$1f9a16440fb0b423;
$d1ef67c10ada5672$export$43167351f6db920a = "_index__progressBarWrapper";
$d1ef67c10ada5672$export$9d1298853f0e77f7 = "_index__progressBar";
$d1ef67c10ada5672$export$efb91eddb63d470a = "_index__sliderBackground";
$d1ef67c10ada5672$export$b38bf4f33e9e4c25 = "_index__slider";
$d1ef67c10ada5672$export$238004eea37cd0e4 = "_index__progres";
$d1ef67c10ada5672$export$1f9a16440fb0b423 = "_index__progressBarText";


function $f6fb1b4c7ba52d3c$export$c17561cb55d4db30({ text: text = 'Loading...' , borderColor: borderColor = 'navy' , sliderColor: sliderColor = 'white' , sliderBackground: sliderBackground = 'red' , id: id = '' , className: className = '' ,  }) {
    const style = `
    .${$d1ef67c10ada5672$export$9d1298853f0e77f7} {
      background-color: ${borderColor};
    }
    .${$d1ef67c10ada5672$export$efb91eddb63d470a} {
      background-color: ${sliderBackground};
    }
    .${$d1ef67c10ada5672$export$b38bf4f33e9e4c25} {
      background-color: ${sliderColor};
    }`;
    return(/*#__PURE__*/ $jKrDH$jsxs("div", {
        id: id,
        className: `${$d1ef67c10ada5672$export$43167351f6db920a} ${className}`,
        children: [
            /*#__PURE__*/ $jKrDH$jsx("div", {
                className: $d1ef67c10ada5672$export$9d1298853f0e77f7,
                children: /*#__PURE__*/ $jKrDH$jsx("div", {
                    className: $d1ef67c10ada5672$export$efb91eddb63d470a,
                    children: /*#__PURE__*/ $jKrDH$jsx("div", {
                        className: $d1ef67c10ada5672$export$b38bf4f33e9e4c25
                    })
                })
            }),
            /*#__PURE__*/ $jKrDH$jsx("div", {
                className: $d1ef67c10ada5672$export$1f9a16440fb0b423,
                children: text
            }),
            /*#__PURE__*/ $jKrDH$jsx("style", {
                children: style
            })
        ]
    }));
}



var $1126dd0ffcca1dcc$export$d043ff22bb17d686;
var $1126dd0ffcca1dcc$export$641374ffb95bc399;
var $1126dd0ffcca1dcc$export$a179c88182cb9d56;
var $1126dd0ffcca1dcc$export$bb628a54ab399bc9;
var $1126dd0ffcca1dcc$export$e1d786d2f707b414;
var $1126dd0ffcca1dcc$export$eb03490a10c3abc9;
var $1126dd0ffcca1dcc$export$6f5fc7f68f30286;
$1126dd0ffcca1dcc$export$d043ff22bb17d686 = "_index__spinnerWrapper";
$1126dd0ffcca1dcc$export$641374ffb95bc399 = "_index__spinner";
$1126dd0ffcca1dcc$export$a179c88182cb9d56 = "_index__viewBox";
$1126dd0ffcca1dcc$export$bb628a54ab399bc9 = "_index__rotate";
$1126dd0ffcca1dcc$export$e1d786d2f707b414 = "_index__circle";
$1126dd0ffcca1dcc$export$eb03490a10c3abc9 = "_index__dash";
$1126dd0ffcca1dcc$export$6f5fc7f68f30286 = "_index__spinnerText";


function $a4c01634f41ac8d6$export$7f7cbe89f1eacd2({ id: id = '' , className: className = '' , color1: color1 = 'green' , color2: color2 = 'lightGreen' , text: text = 'Loading...' , textColor: textColor = '' ,  }) {
    let style = `.${$1126dd0ffcca1dcc$export$6f5fc7f68f30286} {
    background: linear-gradient(90deg, ${color1}, ${color2});
    background-clip: text;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    font-weight: bold;
  }`;
    if (textColor !== '') style = `
      .${$1126dd0ffcca1dcc$export$6f5fc7f68f30286} {
      font-weight: bold;
      color: ${textColor}
      }`;
    return(/*#__PURE__*/ $jKrDH$jsxs("div", {
        id: id,
        className: `${$1126dd0ffcca1dcc$export$d043ff22bb17d686} ${className}`,
        children: [
            /*#__PURE__*/ $jKrDH$jsxs("div", {
                className: $1126dd0ffcca1dcc$export$641374ffb95bc399,
                children: [
                    /*#__PURE__*/ $jKrDH$jsxs("svg", {
                        className: $1126dd0ffcca1dcc$export$a179c88182cb9d56,
                        viewBox: "0 0 50 50",
                        children: [
                            /*#__PURE__*/ $jKrDH$jsx("circle", {
                                className: $1126dd0ffcca1dcc$export$e1d786d2f707b414,
                                cx: "25",
                                cy: "25",
                                r: "22.5",
                                fill: "none",
                                strokeWidth: "5",
                                stroke: "url(#myGradient)"
                            }),
                            /*#__PURE__*/ $jKrDH$jsx("defs", {
                                children: /*#__PURE__*/ $jKrDH$jsxs("linearGradient", {
                                    id: "myGradient",
                                    children: [
                                        /*#__PURE__*/ $jKrDH$jsx("stop", {
                                            offset: "0%",
                                            stopColor: color1
                                        }),
                                        /*#__PURE__*/ $jKrDH$jsx("stop", {
                                            offset: "100%",
                                            stopColor: color2
                                        })
                                    ]
                                })
                            })
                        ]
                    }),
                    /*#__PURE__*/ $jKrDH$jsx("div", {
                        className: `gradientText ${$1126dd0ffcca1dcc$export$6f5fc7f68f30286}`,
                        children: text
                    })
                ]
            }),
            /*#__PURE__*/ $jKrDH$jsx("style", {
                children: style
            })
        ]
    }));
}



var $c618e77364760de4$export$77ff515d9de063ae;
var $c618e77364760de4$export$ba2737e33880d681;
var $c618e77364760de4$export$641374ffb95bc399;
var $c618e77364760de4$export$e1d786d2f707b414;
var $c618e77364760de4$export$ecad260a8a5fef4f;
var $c618e77364760de4$export$71d7c553ee594ba2;
$c618e77364760de4$export$77ff515d9de063ae = "_index__trinitySpinnerWrapper";
$c618e77364760de4$export$ba2737e33880d681 = "_index__trinitySpinner";
$c618e77364760de4$export$641374ffb95bc399 = "_index__spinner";
$c618e77364760de4$export$e1d786d2f707b414 = "_index__circle";
$c618e77364760de4$export$ecad260a8a5fef4f = "_index__animation";
$c618e77364760de4$export$71d7c553ee594ba2 = "_index__trinitySpinnerText";


function $5911710163ad0207$export$7bee1434560ae9ef({ color: color = 'red' , text: text = 'Loading...' , id: id = '' , className: className = '' ,  }) {
    const style = `.${$c618e77364760de4$export$71d7c553ee594ba2} { color: ${color} }`;
    return(/*#__PURE__*/ $jKrDH$jsxs("div", {
        id: id,
        className: `${$c618e77364760de4$export$77ff515d9de063ae} ${className}`,
        children: [
            /*#__PURE__*/ $jKrDH$jsxs("div", {
                className: $c618e77364760de4$export$ba2737e33880d681,
                children: [
                    /*#__PURE__*/ $jKrDH$jsxs("svg", {
                        className: $c618e77364760de4$export$641374ffb95bc399,
                        viewBox: "0 0 100 100",
                        children: [
                            /*#__PURE__*/ $jKrDH$jsx("defs", {
                                children: /*#__PURE__*/ $jKrDH$jsx("filter", {
                                    id: "shadow",
                                    children: /*#__PURE__*/ $jKrDH$jsx("feDropShadow", {
                                        dx: "0",
                                        dy: "0",
                                        stdDeviation: "2",
                                        floodColor: "black"
                                    })
                                })
                            }),
                            /*#__PURE__*/ $jKrDH$jsx("circle", {
                                className: $c618e77364760de4$export$e1d786d2f707b414,
                                style: {
                                    fill: 'transparent',
                                    stroke: color,
                                    strokeWidth: '7px',
                                    strokeLinecap: 'round',
                                    filter: 'url(#shadow)'
                                },
                                cx: "50",
                                cy: "50",
                                r: "45"
                            })
                        ]
                    }),
                    /*#__PURE__*/ $jKrDH$jsx("div", {
                        className: $c618e77364760de4$export$71d7c553ee594ba2,
                        children: text
                    })
                ]
            }),
            /*#__PURE__*/ $jKrDH$jsx("style", {
                children: style
            })
        ]
    }));
}



var $d97c975fb6e8ea7e$export$296d15118585a768;
var $d97c975fb6e8ea7e$export$13c70a3bcea7ce2;
var $d97c975fb6e8ea7e$export$3af1cb7633944de2;
var $d97c975fb6e8ea7e$export$6c4586e4e8bd4b48;
$d97c975fb6e8ea7e$export$296d15118585a768 = "_index__wavesWrapper";
$d97c975fb6e8ea7e$export$13c70a3bcea7ce2 = "_index__waves";
$d97c975fb6e8ea7e$export$3af1cb7633944de2 = "_index__wave";
$d97c975fb6e8ea7e$export$6c4586e4e8bd4b48 = "_index__wavesText";


function $4e976ce5f41316be$export$3f008f9f514cfb66({ backgroundColor: backgroundColor = '#000' , waveColor: waveColor = 'cyan' , text: text = 'Loading...' , id: id = '' , className: className = '' ,  }) {
    const style = `
    .${$d97c975fb6e8ea7e$export$13c70a3bcea7ce2} {
      background-color: ${backgroundColor}
    }
    .${$d97c975fb6e8ea7e$export$3af1cb7633944de2} {
      background: linear-gradient(45deg, ${waveColor}, #fff)
    }`;
    // Setup array of wave elements
    const waves = new Array(10).fill('').map((e, index)=>{
        return(/*#__PURE__*/ $jKrDH$jsx("div", {
            className: $d97c975fb6e8ea7e$export$3af1cb7633944de2,
            style: {
                animationDelay: `0.${index}s`
            }
        }, index));
    });
    return(/*#__PURE__*/ $jKrDH$jsxs("div", {
        id: id,
        className: `${$d97c975fb6e8ea7e$export$296d15118585a768} ${className}`,
        children: [
            /*#__PURE__*/ $jKrDH$jsx("div", {
                className: $d97c975fb6e8ea7e$export$13c70a3bcea7ce2,
                children: waves
            }),
            /*#__PURE__*/ $jKrDH$jsx("div", {
                className: $d97c975fb6e8ea7e$export$6c4586e4e8bd4b48,
                children: text
            }),
            /*#__PURE__*/ $jKrDH$jsx("style", {
                children: style
            })
        ]
    }));
}




export {$f2353fd21d89f897$export$8da50d3722682be0 as Dots, $763dec067a0c5467$export$7248a4fc0dcfadde as MinimalSpinner, $f6fb1b4c7ba52d3c$export$c17561cb55d4db30 as ProgressBar, $a4c01634f41ac8d6$export$7f7cbe89f1eacd2 as Spinner, $5911710163ad0207$export$7bee1434560ae9ef as TrinitySpinner, $4e976ce5f41316be$export$3f008f9f514cfb66 as Waves};
